### ISSUE TITLE
react-hooks/exhaustive-deps rules places error on a different line than eslint

### ISSUE THREAD

**Issue by samsternatmiter** [oxc-project/oxc#18325]
Here's an example of some offending code that I had in my ESLint-ed codebase and already disabled with a `eslint-disable-next-line`:

```tsx
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const deductionCodesMap = integration.connection?.metadata?.employee_navigator?.deductionCodesMap || {};
  return useMemo(() => {
    return Object.keys(deductionCodesMap)
      .map((code) => ({ code, benefitType: deductionCodesMap[code] }))
      .sort((a, b) => {
        return a.code.localeCompare(b.code);
      });
  }, [deductionCodesMap]);
```

Running `oxlint` gives me this warning:
```
  ⚠ eslint-plugin-react-hooks(exhaustive-deps): React hook useMemo depends on `deductionCodesMap`, which changes every render
    ╭─[dashboard/src/pages/integrations/IntegrationConfigs/EmployeeNavigator/EmployeeNavigatorConfig.tsx:35:7]
 27 │   // eslint-disable-next-line react-hooks/exhaustive-deps
 28 │   const deductionCodesMap = integration.connection?.metadata?.employee_navigator?.deductionCodesMap || {};
    ·         ────────┬────────
    ·                 ╰── `deductionCodesMap` is declared here
 29 │   return useMemo(() => {
 30 │     return Object.keys(deductionCodesMap)
 31 │       .map((code) => ({ code, benefitType: deductionCodesMap[code] }))
 32 │       .sort((a, b) => {
 33 │         return a.code.localeCompare(b.code);
 34 │       });
 35 │   }, [deductionCodesMap]);
    ·       ────────┬────────
    ·               ╰── it will always cause this hook to re-evaluate
 36 │ };
    ╰────
  help: Try memoizing this variable with `useRef` or `useCallback`.
```

To disable it, I need this code:
```tsx
const useDeductionCodes = (integration: IntegrationConfigProps["integration"]) => {
  // auto-generated please fix me
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const deductionCodesMap = integration.connection?.metadata?.employee_navigator?.deductionCodesMap || {};
  return useMemo(() => {
    return Object.keys(deductionCodesMap)
      .map((code) => ({ code, benefitType: deductionCodesMap[code] }))
      .sort((a, b) => {
        return a.code.localeCompare(b.code);
      });
    // oxlint-disable-next-line react-hooks/exhaustive-deps
  }, [deductionCodesMap]);
```

This is just a mild inconvenience, but it does mean I need more of these ugly comments in my codebase and it would be great if the two linters found the violation on the same line!

**Comment by samsternatmiter** [comment]
Closing to file properly.
